### PR TITLE
Dev/version in windows dlls

### PR DESCRIPTION
**Pull request by bp-kelley**
Add version information to windows DLLs

### CODE REVIEW

File: Code/cmake/Modules/generate_msvc_version.cmake [hunk 1]
```diff
@@ -0,0 +1,107 @@
+include (CMakeParseArguments)
```
Comment (bp-kelley) [inline Code/cmake/Modules/generate_msvc_version.cmake]: I should add the stack overflow where I got these recipes

Comment (greglandrum) [reply Code/cmake/Modules/generate_msvc_version.cmake]: yes please

**Comment by bp-kelley** [comment]
I've tested this on the c# dlls, but I'll try and do the full suite this weekend.

**Comment by greglandrum** [comment]
Closing due to lack of activity. Feel free to reopen if anyone has the time and inclination to revive this one.

**Pull request closed without merging**